### PR TITLE
Use the providerKey for loading the intercom script

### PR DESCRIPTION
**Pull request by stramel**

### CODE REVIEW

File: src/providers/intercom.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ const load = ({
   providerKey: string
   setState: (state: State) => void
 }): boolean => {
-  const loaded = loadScript()
+  const loaded = loadScript(providerKey)
```
Comment (mikedijkstra) [inline src/providers/intercom.ts:55]: @stramel With this change can remove passing the `app_id` to the `boot` call a few lines below?

Comment (stramel) [reply src/providers/intercom.ts:55]: I don't believe so?

Comment (stramel) [reply src/providers/intercom.ts:55]: Easiest way to remove the setting of `app_id` is to require it in the settings.

`window.intercomSettings = { appId: '3qmk5gyg' };`

Comment (mikedijkstra) [reply src/providers/intercom.ts:55]: I wondered if passing the `app_id` in the URL would set it, but it just seems to redirect. I don't see any difference in what's served if I hit https://widget.intercom.io/widgets/ or https://widget.intercom.io/widgets/otpo7g1i

Do you know if the app ID does anything or could we just drop it from the URL completely?

Comment (stramel) [reply src/providers/intercom.ts:55]: I'm not sure, unfortunately... I know that Intercom recommends using the `app_id` in the URL. I believe it's probably for tracking purposes.

**Comment by mikedijkstra** [comment]
@all-contributors add @stramel for code

**Pull request merged**